gpiozero/pins/__init__.py

```diff
@@ -6,39 +8,133 @@
     )
 str = type('')
 
-import io
-
-from .data import pi_info
 from ..exc import (
     PinInvalidFunction,
     PinSetInput,
     PinFixedPull,
+    PinSPIUnsupported,
     PinPWMUnsupported,
     PinEdgeDetectUnsupported,
+    SPIFixedClockMode,
+    SPIFixedBitOrder,
+    SPIFixedSelect,
+    SPIFixedWordSize,
     )
 
 
-PINS_CLEANUP = []
-def _pins_shutdown():
-    for routine in PINS_CLEANUP:
-        routine()
+class Factory(object):
+    """
+    Generates pins, SPI, and I2C interfaces for devices. This is an abstract
+    base class for pin factories. Descendents must override:
+
+    * :meth:`_get_address`
+    * :meth:`pin_address`
```

Comment:
Oh right ... erm, not sure really. I think keeping getters and setters together is generally a good idea and there's cases where a getter is mandatory but the corresponding setter is optional. Would you favour breaking them up in that case?